front/src/common/Selector/Selector.tsx

```diff
@@ -0,0 +1,161 @@
+import cx from 'classnames';
+import { isNull } from 'lodash';
+import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
+import React, { useState, type PropsWithChildren } from 'react';
+import { GoTrash } from 'react-icons/go';
+
+import SelectNewItemButton from 'common/Selector/SelectNewItemButton';
+
+const DEFAULT_SELECTORS_CLASSNAME = 'selector';
+
+const Selector = <
+  T extends string | null,
+  O extends
+    | NonNullable<T>
+    | {
+        id: NonNullable<T>;
+        label: string;
+      }
+  // All of these 3 properties have to be defined for the extra column to work so we pass them in an object
+>(
+  props: PropsWithChildren<{
+    mainClass?: string;
+    borderClass?: string;
+    title: string;
+    itemsList: { id: T; label: string }[];
+    permanentItems?: T[];
+    selectedItem?: T;
+    onItemSelected?: (item: T) => void;
+    onItemHovered?: (value: string | null) => void;
```

Comment:
is it not T here too ?

```suggestion
    onItemHovered?: (value: T) => void;
```

Reply:
We can use the T type here, and move the logic outside :+1: